extensions/typescript-language-features/src/languageFeatures/tsconfig.ts

```diff
@@ -110,6 +114,55 @@ class TsconfigLinkProvider implements vscode.DocumentLinkProvider {
 	}
 }
 
+// Reference: https://github.com/microsoft/TypeScript/blob/febfd442cdba343771f478cf433b0892f213ad2f/src/compiler/commandLineParser.ts#L3005
+/**
+ * @returns Returns undefined in case of lack of result while trying to resolve from node_modules
+ */
+const getTsconfigPath = async (baseDirUri: vscode.Uri, extendsValue: string): Promise<vscode.Uri | undefined> => {
+	// Don't take into account a case, where tsconfig might be resolved from the root (see the reference)
+	// e.g. C:/projects/shared-tsconfig/tsconfig.json (note that C: prefix is optional)
+
+	const isRelativePath = ['./', '../'].some(str => extendsValue.startsWith(str));
+	if (isRelativePath) {
+		const absolutePath = vscode.Uri.joinPath(baseDirUri, extendsValue);
+		if (await exists(absolutePath)) { return absolutePath; }
+		// Will suggest to create a .json variant if it doesn't exist yet
+		return absolutePath.with({
+			path: `${absolutePath.path}.json`
+		});
+	}
+
+	// Otherwise resolve like a module
+	return resolveNodeModulesPath([
+		extendsValue,
+		`${extendsValue}.json`,
+		`${extendsValue}/tsconfig.json`,
```

Comment:
We should be able to skip some of these checks entirely. For example if the file already ends in `.json`, there is no purpose to checking `${extendsValue}.json` or `${extendsValue}/tsconfig.json`

Failing these early means we can skip going to disk to check if the file exists

Reply:
But in case if we have `module.json/tsconfig.json` in `node_modules` TS can resolving it by referencing `module.json` from extends.
I'm gonna think about it. I'm probably will be there in a day.

Reply:
I made impl a little more complicated, now we skip checks for just `node_modules` and check instead full package path. If the package is not present (with structure from https://github.com/microsoft/vscode/pull/141062#issuecomment-1017432148), it won't check all path candidates within first encountered node_modules and instead 5 checks overall will be performed.
Do you have any other ideas on how it can be optimized or on how to make it cleaner?

Reply:
I wish we had a way to reuse Node's resolving algorithm from somewhere else.